amethyst_assets/src/error.rs

```diff
@@ -1,173 +1,52 @@
 use std::error::Error;
-use std::fmt::{Debug, Display, Formatter, Result as FmtResult};
+use std::fmt::{Display, Formatter, Result as FmtResult};
 
-use futures::future::SharedError;
-
-use asset::AssetSpec;
+use BoxedErr;
 
 /// Error type returned when loading an asset.
-/// Includes the `AssetSpec` and the error (`LoadError`).
+/// Includes
+///
+/// * the `name` of the asset,
+/// * the `format` identifier and
+/// * and the error that occurred during loading.
 #[derive(Debug)]
-pub struct AssetError<A, F, S> {
+pub struct AssetError {
     /// The specifier of the asset which failed to load
-    pub asset: AssetSpec,
+    pub name: String,
+    /// The format identifier.
+    pub format: String,
     /// The error that's been raised.
-    pub error: LoadError<A, F, S>,
+    pub error: BoxedErr,
 }
 
-impl<A, F, S> AssetError<A, F, S> {
-    pub(crate) fn new(asset: AssetSpec, error: LoadError<A, F, S>) -> Self {
-        AssetError { asset, error }
+impl AssetError {
+    pub(crate) fn new(name: String, format: String, error: BoxedErr) -> Self {
+        AssetError {
+            name,
+            format,
+            error,
+        }
     }
 }
 
-impl<A, F, S> Display for AssetError<A, F, S>
-where
-    A: Display,
-    F: Display,
-    S: Display,
-{
+impl Display for AssetError {
     fn fmt(&self, f: &mut Formatter) -> FmtResult {
         write!(
             f,
-            "Failed to load asset \"{}\" of format \"{:?}\" from storage with id \"{}\": {}",
-            &self.asset.name,
-            &self.asset.exts,
-            &self.asset.store.id(),
+            "Failed to load asset \"{}\" of format \"{}\": {}",
```

Comment:
Pssst, the " character is automatically added to the beginning and end if you print a string with Debug.  Note this is not a requested change, just a note for future code.